CRM/Odoosync/Hook/Post/Contact/Base.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * Abstraction class provides skeleton for contact post processes
+ */
+abstract class CRM_Odoosync_Hook_Post_Contact_Base extends CRM_Odoosync_Hook_Post_Base {
+
+  /**
+   * Updates contact sync information
+   */
+  public function process() {
+    if (CRM_Odoosync_Hook_Post_Contact_Checker::isContactSyncInfoUpdated()) {
```

Comment:
I think there is an issue with this implementation, suppose you added a contact with an address, the post hook will get called for the address entity before the contact entity , but  CRM_Odoosync_Hook_Post_Contact_Entity class always use "update" for the sync operation in the database but the it should be "create" instead. it might be that what I said is not true and I am missing something so please double-check that.

Reply:
Yes, you are right. “Contact sub entity hook” called before “Contact entity hook“ and if the Contact was created then ‘Action to sync” was set as - "Update" instead of "Create". 

We’ve fixed it. And as a result  we removed code which prevents many of “Sync Contact Info” updates. So now when the Contact is created, and “Contact sub entity hook” was called early
than “Contact entity hook“ then “Contact sub entity hook” replaces ‘Create’ for ‘Update’ in ‘Action to sync”. (“Contact entity hook“ always calls after “Contact sub entity hook’.)

Reply:
well, there is a better alternative probably rather than removing that code which is useing pre and post hooks togther, the order in pre hook is correct so we can dedect if the operation is contact creation in it and store a static boolean flag in the handler class to indicate that and then use that flag in the post hook.

But I think the current solution is still fine, I don't want to over-complicate things for now.